infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -97,25 +101,92 @@ public LogReplicationMetadataManager(CorfuRuntime runtime, LogReplicationContext
+    private static Table<ReplicationEventInfoKey, ReplicationEvent, Message>
+    tryOpenReplicationEventTable(CorfuStore corfuStore) {
+        Table<ReplicationEventInfoKey, ReplicationEvent, Message> table = null;
+        try {
+            table = corfuStore.openTable(NAMESPACE, REPLICATION_EVENT_TABLE_NAME,
                     ReplicationEventInfoKey.class,
                     ReplicationEvent.class,
                     null,
-                    TableOptions.fromProtoSchema(ReplicationEvent.class));
-        } catch (Exception e) {
-            log.error("Caught an exception while opening metadata tables", e);
-            throw new ReplicationWriterException(e);
+                    TableOptions.fromProtoSchema(ReplicationMetadata.class));
+        } catch (Exception ee) {
+            log.error("Caught an exception while opening replication event table", ee);
+            throw new ReplicationWriterException(ee);
         }
+        return table;
+    }
+
+    /**
+     * The following tables need to move from old type to new type
+     * 1. CORFU-REPLICATION-WRITER-UUID-OF-CLUSTER -> LogReplicationMetadataTable
+     * Type: Schema change AND name change
+     *    (LogReplicationMetadataKey, LogReplicationMetadataVal) ->
+     *     (LogReplicationSession, ReplicationMetadata)
+     * Action: Open the old table with old types and Clear the old table
+     *         Not used in new code
+     *
+     * 2. LogReplicationStatus
+     * Type: Schema change for key AND value types with NO table name change
+     * (ReplicationStatusKey, ReplicationStatusVal) ->
+     * (LogReplicationSession.class, ReplicationMetadata.class)
+     * Action: Make serializer aware of old types, open table with new type, clear it
+     *
+     * 3. LogReplicationEventTable
+     * Type: Schema change for Key type only
+     * (ReplicationEventKey, ReplicationEvent) ->
+     * (ReplicationEventInfoKey, ReplicationEvent)
+     * Action: Make serializer aware of old key type, open table with new type, clear it
+     * @param txnContext - the parent transaction where check for rolling upgrade completion occurs
+     */
+    public static void migrateData(CorfuStore corfuStore, TxnContext txnContext) {
+        final ExecutorService executorService = Executors.newSingleThreadExecutor();
+        corfuStore.getRuntime().getTableRegistry()
+                .addTypeToClassMap(LogReplicationMetadata.ReplicationStatusKey.getDefaultInstance());
+        corfuStore.getRuntime().getTableRegistry()
+                .addTypeToClassMap(LogReplicationMetadata.ReplicationStatusVal.getDefaultInstance());
+        corfuStore.getRuntime().getTableRegistry()
+                .addTypeToClassMap(LogReplicationMetadata.ReplicationEventKey.getDefaultInstance());
+        executorService.submit(() -> {
+            tryOpenReplicationEventTable(corfuStore);
+            tryOpenReplicationEventTable(corfuStore);
```

Comment:
This line seems to be a duplicate, should it be `tryOpenReplicationStatusTable()`? Also we aren't handling the first scenario with the old metadata table here will that be something we can add a todo for, or handle elsewhere?